presto-orc/src/main/java/com/facebook/presto/orc/writer/CompressionBufferPool.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.writer;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import static io.airlift.slice.SizeOf.sizeOf;
+
+public interface CompressionBufferPool
+{
+    byte[] checkOut(int length);
+
+    void checkIn(byte[] buffer);
+
+    long getRetainedBytes();
+
+    class LastUsedCompressionBufferPool
```

Comment:
I think common name for this kind of things is `Most/Least Recently Used Bla`. But in this case we have a pool of a single element, so idk.

Do we have any numbers on the max size and hit rate?

Reply:
I thought a lot about the name, most of them indicates that it is a cache with different keys. But what we want is where only one buffer needs to be remembers. I can change it to MostRecentlyUsedCompressionBuffer if it sounds better.